app/models/product.rb

```diff
@@ -22,7 +22,8 @@ class Product < ApplicationRecord
   has_many :prices, as: :priceable, dependent: :destroy
   has_many :categories_by_prices, through: :prices, source: :category
 
-  validates :title, presence: true, length: { in: 2..50 }
+  validates :title, presence: true, uniqueness: true, format: { with: /\A[a-zA-Zа-яієїґ'А-ЯІЄЇҐ0-9-\s]+\z/ }
+  validates :title, length: { in: 2..30 }, if: -> { title.present? }
```

Comment:
У тебе валідація на присутність, тому нема сенсу в тому іфі)
```suggestion
  validates :title, presence: true, uniqueness: true, format: { with: /\A[a-zA-Zа-яієїґ'А-ЯІЄЇҐ0-9-\s]+\z/ }, length: { in: 2..30 }
```

Reply:
зробив перший рівнем валідації
validates :title, presence: true, uniqueness: true

а далі якщо title.present?
format: { with: /\A[a-zA-Zа-яієїґ'А-ЯІЄЇҐ0-9-\s]+\z/ }, length: { in: 2..30 }

бо інакше якщо намагатись створити з пустим то буде 3 помилки
Title can't be blank, Title contains invalid characters, and Title too short (minimum 2 symbols)
то як на мене не є гуд для просто пустої назви